tools/codegen/gen.py

```diff
@@ -454,6 +461,12 @@ def gen_one(f: NativeFunction) -> Optional[str]:
 }}
 
 }} // anonymous namespace
+
+namespace {self.dispatch_key.lower()} {{
```

Comment:
Probably not a big deal, but the individual `namespace cpu {...}` for every op will probably make these files a few thousand lines longer, vs. grouping them all together in one namespace block

Reply:
Yeah. The alternative is to split up the implementations in codegen. I'm ambivalent about this; so if someone feels strongly I'll swap it around.

> Actually, isn't this generating at::<dispatch_key> functions for every dispatch key, and then only providing headers for the specific keys we want (cpu/cuda)?

Technically yes, but in reality only CPU and CUDA are supported by structured, so there isn't actually any wastage.

Reply:
> Yeah. The alternative is to split up the implementations in codegen. I'm ambivalent about this; so if someone feels strongly I'll swap it around.

Thought it was worth calling out, but I'm ambivalent as well :)

> Technically yes, but in reality only CPU and CUDA are supported by structured, so there isn't actually any wastage.

Ah right, yeah